### PR TITLE
Parallelization option for daily and multi--variable data 

### PR DESCRIPTION
**Pull request by eshchekinova**

### CODE REVIEW

File: motupydownload/motupydownload.py [hunk 1]
```diff
@@ -122,34 +165,24 @@
...
-    # call motuclient for every variable and write into daily output file
-    for variable_name in variables:
-        for day in range(num_days):
-            # set time limits corresponding to start_time and end_time in string format
-            start_time = (time_start + timedelta(day)).strftime("%Y-%m-%d")
-            end_time = (time_start + timedelta(day + 1)).strftime("%Y-%m-%d")
-            # set time limits for motuclient in format "YYYY-mm-dd ss:MM:HH"
-            time_left_motu = f"{start_time}T00:00:00"
-            time_right_motu = f"{end_time}T00:00:00"
+    num_cores = multiprocessing.cpu_count()
```
Comment (willirath) [inline motupydownload/motupydownload.py]: I'd rather use a command line flag for setting the number of parallel downloads:
- We're not really using CPU anyway (time is determined by the download bandwidht).
- On large systems like the Nesh frontends, `cpu_count()` might be very large (48 or 64 or 96). This is probably higher than the number of parallel downloads allowed by the MOTU server.

**Review by willirath — CHANGES_REQUESTED** [review]

File: motupydownload/motupydownload.py (hunk 1 above)
Comment (eshchekinova) [reply motupydownload/motupydownload.py]: Thank you for comments, last week you showed me one way to parallelize the loop, could you send me the link to look it up again.

Comment (willirath) [reply motupydownload/motupydownload.py]: I've suggested to use `multiprocessing.Pool.map` as described here: https://docs.python.org/3/library/multiprocessing.html The advantage of using `multiprocessing`, which is part of the Python standard library, would be avoiding an additional dependency.

**Review by eshchekinova — COMMENTED** [review]
Approved

**Review by willirath — CHANGES_REQUESTED** [review]
Let's add a command line flag that sets the number of parallel downloads. I think it's a risky idea to have a `Pool` of size `number_days x variables`. Instead, let's try something like
```shell
parser.add_argument(
    "--parallel-downlodas",
    default=8,
    help=("Number of parallel downloads. Defaults to 8."),
)
```
(untested)

**Review by eshchekinova — COMMENTED** [review]
Reviewed

**Review by eshchekinova — COMMENTED** [review]
The longitude maximum is corrected

**Review by willirath — APPROVED** [review]

**Comment by willirath** [comment]
I've done some cleanup in https://github.com/geomar-od/rasmus-cmems-downloads/pull/27/commits/3a692896645e6a5d8aed0ec7fb2379e35af58972 and resolved conflicts with `master`.

Merging.

**Pull request merged**